### PR TITLE
adapt atlas_proxy to new model of amundsenatlastypes==1.0.2

### PR DESCRIPTION
**Pull request by dwarszawski**
### Summary of Changes

adapt atlas_proxy to [amundsenatlastypes](https://github.com/dwarszawski/amundsen-atlas-types) == 1.0.2 

### Tests

modified test_atlas_proxy accordingly

### Documentation

_What documentation did you add or modify and why? Add any relevant links then remove this line_

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [ ] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [ ] PR includes a summary of changes. 
- [ ] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does
- [ ] PR passes `make test`

### CODE REVIEW

**Comment by feng-tao** [comment]
@dwarszawski the CLA check is failing, could you help to fix(normally due to github handle not config locally)?

also cc @verdan

**Comment by dwarszawski** [comment]
> @dwarszawski the CLA check is failing, could you help to fix(normally due to github handle not config locally)?
> 
> also cc @verdan

I just signed the CLA, but still failing on check.

**Comment by feng-tao** [comment]
@dwarszawski could you rebase your pr with master if you haven't? thanks.

**Comment by feng-tao** [comment]
Fix is merged, let me trigger the test in your pr.

File: docs/proxy/atlas/popular_tables.md [hunk 1]
```diff
@@ -6,82 +6,44 @@ The required entity definitions for Atlas can be applied using [amundsenatlastyp
...
+            "DataSet"
+          ],
+          "attributeDefs": [
+            {
+              "name": "popularityScore",
+              "typeName": "float",
+              "isOptional": true,
+              "cardinality": "SINGLE",
+              "isUnique": false,
+              "isIndexable": false,
```
Comment (verdan) [inline docs/proxy/atlas/popular_tables.md:33]: why this value is `false` ? shouldn't this be indexable ?

File: metadata_service/proxy/atlas_proxy.py [hunk 2]
```diff
@@ -182,11 +179,12 @@ def _create_reader(self, metadata_guid: str, user_guid: str, reader_qn: str, tab
...
                            'isFollowing': True,
                            'count': 0,
-                           'entityMetadata': {'guid': metadata_guid},
-                           'user': {'guid': user_guid},
-                           'entityUri': table_uri}
+                           'user': {'guid': user_guid}}
         }
-        self._driver.entity_bulk.create(data={'entities': [reader_entity]})
+
+        readers = table_entity.entity[self.ATTRS_KEY]['readers'].append(reader_entity)
```
Comment (verdan) [inline metadata_service/proxy/atlas_proxy.py:185]: Are you making sure the uniqueness of relation between a user and a table reader ?

File: metadata_service/proxy/atlas_proxy.py [hunk 3]
```diff
@@ -182,11 +179,12 @@ def _create_reader(self, metadata_guid: str, user_guid: str, reader_qn: str, tab
...
+        table_entity.entity[self.ATTRS_KEY]['readers'] = readers
+        table_entity.update()
```
Comment (verdan) [inline metadata_service/proxy/atlas_proxy.py:187]: Does this create a new reader entity as well ? (just curious is you need to create a reader entity first and then update in table readers.

File: metadata_service/proxy/atlas_proxy.py [hunk 4]
```diff
@@ -444,46 +436,29 @@ def get_column_description(self, *,
...
                                 'excludeDeletedEntities': True,
-                                'limit': num_entries,
-                                'attributes': ['table']}
+                                'limit': num_entries}
+
+        table_entities: List = list()
+
+        try:
+            search_results = self._driver.search_basic.create(data=popular_query_params)
+            for table in search_results.entities:
```
Comment (verdan) [inline metadata_service/proxy/atlas_proxy.py:455]: Why are we looping over entities here ? You could simply use `search_results.entities` directly on line 463.

Comment (dwarszawski) [reply metadata_service/proxy/atlas_proxy.py]: Agree, that try/except block seems to be redundant now. Removed this additional loop.

File: metadata_service/proxy/atlas_proxy.py (hunk 3 above)
Comment (dwarszawski) [reply metadata_service/proxy/atlas_proxy.py:187]: You are right, encounter this problem while generating the test data - reader entity must be first created then appended. Fixed in the latest commit.

File: metadata_service/proxy/atlas_proxy.py (hunk 2 above)
Comment (dwarszawski) [reply metadata_service/proxy/atlas_proxy.py]: this should be handled by atlas if reader entity is created first.

File: docs/proxy/atlas/popular_tables.md (hunk 1 above)
Comment (dwarszawski) [reply docs/proxy/atlas/popular_tables.md:33]: This is not used as a predicate but only for sorting. Introducing the index would probably improve the sort perfomance but may slow down write operations. Let's follow up on this discussion here https://github.com/dwarszawski/amundsen-atlas-types

**Comment by feng-tao** [comment]
@verdan @dwarszawski are we good for the latest version of the pr?

**Comment by verdan** [comment]
Looks good to me ! Thank you @dwarszawski. 👍 💯

**Pull request merged**